GoogleForCreators/web-stories-wp
# Dashboard: Add Post Lock UI to List View

## Context

Realized that the post lock experiment never got added to the list view in the dashboard. 

## Summary

Adds indicator of a story being locked by another user in the editor to the list view.

![Screen Shot 2021-07-26 at 3 24 28 PM](https://user-images.githubusercontent.com/10720454/127066982-d030cb70-3da2-4209-bff9-386d666fd3d2.png)


## Relevant Technical Choices

- Gave the `MyStories` list view it's own styled components file(s) - one for the table(list) and another for each row(item).
- Moved the table rows to its own memoized component  
- Clean up some text styles with the design system that could be simplified, otherwise the only new style added is the locked icon + tooltip, everything else is just moving and reorganizing. 


## To-do

- [x] Part of this will disappear once I merge #8397 since the story menu generator is updated the same way in that. 

## User-facing changes

- If post lock experiment is not enabled, no changes 
- With post lock experiment enabled you'll now see a lock icon when another user is editing a story with a tooltip specifying who's editing the story. For keyboard users, the title is given a new aria-label when a story is locked so it's announced. 

## Testing Instructions

### QA




This PR can be tested by following these steps:

- First make sure list view on dashboard works as expected without post lock experiment enabled 
- Now enable the post lock experiment and open a new browser or incognito and log in as another user with a story open in the editor, use your first user and go to the list view, see that there's a lock indication and renaming a story is disabled. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #8423
